src/resources/rmt_esp32.cc

```diff
@@ -362,6 +333,44 @@ PRIMITIVE(transmit_and_receive) {
   return data;
 }
 
+PRIMITIVE(receive) {
+  ARGS(RMTResource, resource, int, max_output_len, int, timeout_ms)
+
+  auto tx_channel = kInvalidChannel;
+  auto rx_channel = resource->channel();
+  const rmt_item32_t* transmit_items = null;
+  const rmt_item32_t* receive_items = null;
+  int transmit_items_length = 0;
+  int receive_items_length = 0;
+  return transmit_and_receive(tx_channel, rx_channel,
+                              transmit_items, transmit_items_length,
+                              receive_items, receive_items_length,
+                              max_output_len,
+                              timeout_ms,
+                              process);
+}
+
+PRIMITIVE(transmit_and_receive) {
+  ARGS(RMTResource, tx, RMTResource, rx, Blob, transmit_bytes, Blob, receive_bytes, int, max_output_len, int, receive_timeout_ms)
+  if (transmit_bytes.length() % 4 != 0) INVALID_ARGUMENT;
+  if (receive_bytes.length() % 4 != 0) INVALID_ARGUMENT;
+
+  rmt_channel_t rx_channel = rx->channel();
+  rmt_channel_t tx_channel = tx->channel();
+
+  const rmt_item32_t* transmit_items = reinterpret_cast<const rmt_item32_t*>(transmit_bytes.address());
+  const rmt_item32_t* receive_items = reinterpret_cast<const rmt_item32_t*>(receive_bytes.address());
+  int transmit_items_length = transmit_bytes.length() / 4;
```

Comment:
Should you pass null items if the lengths are zero?

Reply:
Good find.
Switched to checking for the length in the shared function instead, this way I avoid having a check here.